packages/tees/__test__/src/lifecycle/postSetup.test.js

```diff
@@ -0,0 +1,230 @@
+
+            expect(result.context.driver.run).toBeCalled();
+
+        });
+
+        it('execCase, if test only should assign the params of global', async () => {
+
+            context.isOnly = true;
+
+            result = await getExecCaseParams(context);
+            global.getExecCaseParams = jest.fn().mockReturnValue(result);
+
+            global.beforeEachStart = jest.fn();
+
+            result.context.driver.run = jest.fn();
+            result.context.driver.newPage = jest.fn();
+            result.context.driver.goto = jest.fn();
+
+            await execCase(context);
+
+            expect(result.context.driver.run).toBeCalled();
+        });
+
+        it('execCase, if isUT is true context.driver.run should not be called', async () => {
+
+            context.driver = "ut";
+
+            result = await getExecCaseParams(context);
+
+            global.getExecCaseParams = jest.fn().mockReturnValue(result);
+
+            global.beforeEachStart = jest.fn();
+
+            result.context.driver.run = jest.fn();
+            result.context.driver.newPage = jest.fn();
+            result.context.driver.goto = jest.fn();
+
+            await execCase(context);
+
+            expect(result.context.driver.run).not.toBeCalled();
+
+            context.driver = "puppeteer";
+
+        });
+
+        it('execCase, verify the params of global and global.beforeEachStart should be called', async () => {
+
+            context.isHeadless = true;
+            context.isDebugger = true;
+            context.isVerbose = true;
+
+            result = await getExecCaseParams(context);
+
+            global.getExecCaseParams = jest.fn().mockReturnValue(result);
+
+            global.beforeEachStart = jest.fn();
+
+            result.context.driver.run = jest.fn();
+            result.context.driver.newPage = jest.fn();
+            result.context.driver.goto = jest.fn();
+
+            await execCase(context);
+
+            expect(result.context.driver.run).toBeCalled();
+
+        });
+
+
+        it('execCase, if isSandbox is false context.driver.run should not be called', async () => {
+
+            context.isSandbox = false;
+
+            const result = await getExecCaseParams(context);
+
+            global.getExecCaseParams = jest.fn().mockReturnValue(result);
+
+            global.beforeEachStart = jest.fn();
+
+            result.context.driver.run = jest.fn();
+            result.context.driver.newPage = jest.fn();
+            result.context.driver.goto = jest.fn();
+
+            await execCase(context);
+
+            expect(result.context.driver.run).not.toBeCalled();
+
+        });
+
+    }),
+
+    describe('postSetup unit test : testCase', () => {
+
+        it(`testCase, global.execCase should be called and the param is equal to ${execCaseParamExpected}`, async () => {
+
+            global.execCase = jest.fn();
+
+            await testCase(caseParams, argFn);
```

Comment:
caseParams
1.You can override different scenarios by setting this parameter
for example: mode is null or not null